src/main/java/uk/gov/justice/hmpps/datacompliance/services/migration/OffenderImageUploader.java

```diff
@@ -43,10 +45,19 @@ private void getAndUploadImageData(final OffenderImageMetadata image, final Offe
 
         imageData.ifPresentOrElse(
 
-                data -> imageRecognitionClient.uploadImageToCollection(data, offenderNumber, image.getImageId())
-                        .ifPresent(faceId -> uploadLogger.log(offenderNumber, image, faceId)),
+                data -> uploadAndLogImage(image, offenderNumber, data),
 
                 () -> log.warn("Image: '{}' for offender: '{}' has no image data",
                         image.getImageId(), offenderNumber.getOffenderNumber()));
     }
+
+    private void uploadAndLogImage(final OffenderImageMetadata image,
+                                   final OffenderNumber offenderNumber,
+                                   final byte[] imageData) {
+
+        rateLimiter.acquire();
```

Comment:
So I assume this just blocks until it is allowed to process? This is excellent.

Reply:
Yes, correct